src/main/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceService.scala

```diff
@@ -222,6 +218,30 @@ class WorkspaceService(protected val argUserToken: WithAccessToken, val rawlsDAO
 
   }
 
+  def unPublishSuccessMessage(workspaceNamespace: String, workspaceName: String): String = s" The workspace $workspaceNamespace:$workspaceName has been un-published."
+
+  def deleteWorkspace(ns: String, name: String): Future[PerRequestMessage] = {
+    rawlsDAO.getWorkspace(ns, name) flatMap { ws =>
+      val wsPublished: Boolean = isPublished(ws)
+      if (wsPublished) {
+        rawlsDAO.updateLibraryAttributes(ns, name, updatePublishAttribute(false)) flatMap { f1 =>
+          Future(removeDocument(ws.workspace, searchDAO)).flatMap { f2 =>
+            rawlsDAO.deleteWorkspace(ns, name) map { f3 =>
+              RequestComplete(f3.copy(message = Some(f3.message.getOrElse("") + unPublishSuccessMessage(ns, name))))
+            }
+          }
+        } recover {
+          case e: FireCloudExceptionWithErrorReport => RequestComplete(e.errorReport.statusCode.getOrElse(InternalServerError), e.errorReport)
+          case e => RequestComplete(InternalServerError, ErrorReport(message = e.getMessage))
+        }
```

Comment:
by replaying the error message from rawls, the end user - in the case where the user is owner but not curator - gets a message "You must be a curator and either be an owner or have catalog with read+." That's correct, but not necessarily helpful. Maybe change to something like `s"You cannot delete this workspace: ${e.getMessage}"`